go-controller/pkg/ovn/egressgw.go

```diff
@@ -392,6 +392,9 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod *kapi.Pod, namespace str
 	if err := oc.deleteGWRoutesForNamespace(namespace, foundGws.gws); err != nil {
 		// add the entry back to nsInfo for retrying later
 		nsInfo, nsUnlock := oc.getNamespaceLocked(namespace, false)
+		if nsInfo == nil {
+			return fmt.Errorf("failed to get nsInfo %s to add back all the gw routes: %w", namespace, err)
+		}
```

Comment:
A few lines above if nsInfo == nil we return nil, but here we return an error. If this indeed should be the case, I am missing some comments that explain.

Reply:
I think the problem is github showing too less lines - this `err != nil` branch ends up returning an error after modifying `nsInfo` like 4 lines lower. So this change just prevents nil dereference and returns without adding entry back.
let me know if you think I can add any comments to the code to explain this better

Reply:
If `oc.deleteGWRoutesForNamespace(namespace, foundGws.gws)` failed, it means cleanup wasn't completed and needs to be retried - so this `if` branch should return error anyway. We can't just hope that whoever deleted nsInfo also did the proper cleanup right?

Reply:
I think returning error is fine here. If the namespace is gone, then when this is retried deleteGWRoutesForNamespace will not return error, and we wouldn't enter this block again.

Reply:
Ah, ok, thanks for explaining, I wasn't seeing the full picture.